Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -1153,7 +1153,7 @@ protected void DiscoverTypes()
             Logger.LogInformation("Loading assemblies");
 
             AllConcreteTypes = GetComposablePartAssemblies()
-                .SelectMany(x => x.ExportedTypes)
+                .SelectMany(x => x.DefinedTypes)
```

Comment:
I'm slightly wary of this change, but I believe you @cvium

Reply:
I knew there would be possible regressions, but in the end `ExportedTypes` forces more "correct" code. You don't really expect a private/internal class to be initialized outside of it's assembly.

Reply:
As I said in chat, I really expected `private` stuff to stay private... and to mark all interfaces we want to be loaded as `public`.
So ideally this change shouldn't be needed.

Reply:
I'm also in favour of making all required classes and interfaces public. It seems more proper.

Reply:
> I'm slightly wary of this change, but I believe you @cvium

I've removed it, but what you all seem to forget is that this was in fact the old behaviour prior to the DI PR 😛 imo, this is such a minor thing compared to everything else that it's not worth the hard-to-debug problems that may arise.

I only knew how to debug this because I encountered it once before and know what the difference is between `DefinedTypes` and `ExportedTypes`. There is no logging to help you in this case.